parsers/rhea/rhea.go

```diff
@@ -0,0 +1,742 @@
+	ContainsN    bool   `json:"containsn" db:"containsn"`
+	Minus        bool   `json:"minus" db:"minus"` // Only set to true if ContainsN == true to handle Nminus1
+	Plus         bool   `json:"plus" db:"plus"`   // Only set to true if ContainsN == true to handle Nplus1
+	Accession    string `json:"reactionParticipant" db:"ReactionParticipant"`
+	Compound     string `json:"compound" db:"compound"`
+}
+
+// Reaction represents a Rhea reaction. Substrates, Products, and SubstrateOrProducts are all ReactionSide accession
+// numbers, which can be linked to the ReactionParticipant's ReactionSide accession
+type Reaction struct {
+	ID                   int      `json:"id" db:"id"`
+	Directional          bool     `json:"directional" db:"directional"`
+	Accession            string   `json:"accession" db:"accession"`
+	Status               string   `json:"status" db:"status"`
+	Comment              string   `json:"comment" db:"comment"`
+	Equation             string   `json:"equation" db:"equation"`
+	HTMLEquation         string   `json:"htmlequation" db:"htmlequation"`
+	IsChemicallyBalanced bool     `json:"ischemicallybalanced" db:"ischemicallybalanced"`
+	IsTransport          bool     `json:"istransport" db:"istransport"`
+	Ec                   string   `json:"ec" db:"ec"`
+	Location             string   `json:"location" db:"location"`
+	Citations            []string `json:"citations"`
+	Substrates           []string `json:"substrates"`
+	Products             []string `json:"products"`
+	SubstrateOrProducts  []string `json:"substrateOrProducts"`
+}
+
+/******************************************************************************
+
+Parse functions
+
+These functions take in the rhea.rdf.gz dump file and return a Rhea struct,
+which contains all of the higher level structs
+
+******************************************************************************/
+
+// ParseRhea parses a list of bytes into a higher-level Rhea Struct.
+func ParseRhea(rheaBytes []byte) (Rhea, error) {
+	var err error
+	// Read rheaBytes into a Rdf object
+	var rdf Rdf
+	err = xml.Unmarshal(rheaBytes, &rdf)
+	if err != nil {
+		return Rhea{}, err
+	}
+
+	// Initialize Rhea
+	var rhea Rhea
+	compoundParticipantMap := make(map[string]string)
+	compoundMap := make(map[string]Compound)
+
+	for _, description := range rdf.Descriptions {
+		// Handle the case of a single compound -> reactive part, such as
+		// <rdf:Description rdf:about="http://rdf.rhea-db.org/Compound_10594">
+		// 	<rh:reactivePart rdf:resource="http://rdf.rhea-db.org/Compound_10594_rp2"/>
+		// </rdf:Description>
+		if (len(description.Subclass) == 0) && (description.ReactivePartXML.Resource != "") {
+			compoundParticipantMap[description.ReactivePartXML.Resource] = description.About
+		}
+		if description.Compound.Resource != "" {
+			compoundParticipantMap[description.About] = description.Compound.Resource
+		}
+
+		for _, subclass := range description.Subclass {
+			switch subclass.Resource {
```

Comment:
factoring the code blocks in this switch statement small individual functions would make this more readable and allow easy unit testing.

Reply:
Normally I like to have larger code blocks rather than a bunch of functions everywhere, but in this particular example, I agree with you. These code blocks are way too large for easy readability. Will factor out.